src/aws-cpp-sdk-core/source/Aws.cpp

```diff
@@ -172,6 +184,20 @@ namespace Aws
 
     void ShutdownAPI(const SDKOptions& options)
     {
+        std::unique_lock<std::mutex> lock(s_initShutdownMutex);
+        if(s_initCount != 1)
+        {
+            if(!s_initCount) {
+                AWS_LOGSTREAM_ERROR(ALLOCATION_TAG, "Unable to ShutdownAPI of AWS-SDK-CPP: the SDK was not initialized.");
```

Comment:
I have no idea how this one is going to be actually traced (without an initialized logger)
I don't want to add an assert here though.